File: back-end/src/controllers/user_controller.rs
use axum::{Json, Router, extract::State, routing::get};

use crate::{
    controllers::extractors::AuthUser, error::AppError, models::auth::UserResponse,
    services::user_service, state::AppState,
};

#[utoipa::path(
    get,
    path = "/users",
    tag = "users",
    responses(
        (status = 200, body = [UserResponse]),
        (status = 401, body = crate::error::ErrorResponse)
    )
)]
pub async fn list_users(
    State(state): State<AppState>,
    _user: AuthUser,
) -> Result<Json<Vec<UserResponse>>, AppError> {
    let users = user_service::list_users(&state).await?;
    Ok(Json(users))
}

pub fn routes() -> Router<AppState> {
    Router::new().route("/users", get(list_users))
}

File: bun_hono/src/controllers/userController.ts
import type { Context } from 'hono';
import type { AppState } from '../config/state';
import { getAuthUser } from '../middleware/auth';
import { listUsers } from '../services/userService';

export const listUsersHandler = (state: AppState) => async (c: Context) => {
  getAuthUser(c);
  const users = await listUsers(state);
  return c.json(users);
};
